src/Test.FunctionalTests.Sarif/MultitoolCommandLineTests.cs

```diff
@@ -35,5 +36,64 @@ public void Multitool_LaunchesAndRunsSuccessfully()
                 process.ExitCode.Should().Be(0);
             }
         }
+
+#if DEBUG
+        [Fact]
+        [Trait(TestTraits.WindowsOnly, "true")]
+        public void Multitool_LaunchesAndRunsSuccessfully_With12kFiles()
+        {
+            using var assertionScope = new AssertionScope();
+
+            string multitoolPath = Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(),
+                @"..\..\Sarif.Multitool\netcoreapp3.1\Sarif.Multitool.exe"));
+
+            string directoryPath = Path.Combine(Path.GetTempPath(), "SarifMultitool12kTestFiles");
+             
+            if (Directory.Exists(directoryPath))
+            {
+                Directory.Delete(directoryPath, true);
+            }
+
+            Directory.CreateDirectory(directoryPath);
+
+            for (int i = 1; i <= 12000; i++)
```

Comment:
Soft-recommend (do not require) using internalsvisibleto to couple this parameter to the channel size.
`ChannelCapacity`.

Signing off.

Reply:
updated to use the ChannelCapacity + 1.
Tried a few times, only need + 1 to repro.